[docs] improves the migration guide (#21159)
expo/expo: docs/pages/eas-update/migrate-to-eas-update.mdx

## Publish an update

After making a change to your project locally, you're ready to publish an update, run:

<Terminal
  cmd={[
    '$ eas update --channel [channel-name] --message [message]',
    '',
    '# Example',
    '',
    '$ eas update --channel production --message "Fixes typo"',
  ]}
  cmdCopy="eas update --channel [channel-name] --message [message]"
/>

Once published, you can see the update in the [Expo dashboard](https://expo.dev/accounts/[account]/projects/[project]/updates).

## Additional migration steps

- Replace instances of `expo publish` with `eas update` in scripts. You can view all the options for publishing with `eas update --help`.
- If you have any code that references `Updates.releaseChannel` from the `expo-updates` library, replace them with `Updates.channel`.
- Remove any code that references `Constants.manifest`. That will now always return `null`. If you're using SDK 46 or above, you can access most properties you'll need with `Constants.expoConfig` from the `expo-constants` library.

## Learn more

> The steps we described above allow you to use a similar flow to Classic Updates. However, EAS Update is more flexible and has more features, which can be used to create more stable release flows. Learn [how EAS Update works](/eas-update/how-eas-update-works) and how you can craft a more stable [deployment process](/eas-update/deployment-patterns) for your project and your team.

Reviewer: ```suggestion
The steps described above allow you to use a similar flow to Classic Updates. However, EAS Update is more flexible and has more features. It can be used to create more stable release flows. Learn [how EAS Update works](/eas-update/how-eas-update-works) and how you can craft a more stable [deployment process](/eas-update/deployment-patterns) for your project and your team.
```

Small suggestion (shouldn't block the PR): Splitting the sentence into multiple can help quickly skim from reader's perspective.

> The steps we described above allow you to use a similar flow to Classic Updates. However, EAS Update is more flexible and has more features, which can be used to create more stable release flows. Learn [how EAS Update works](/eas-update/how-eas-update-works) and how you can craft a more stable [deployment process](/eas-update/deployment-patterns) with your project and your team.

Author: ```suggestion
The steps we described above allow you to use a similar flow to Classic Updates. However, EAS Update is more flexible and has more features, which can be used to create more stable release flows. Learn [how EAS Update works](/eas-update/how-eas-update-works) and how you can craft a more stable [deployment process](/eas-update/deployment-patterns) for your project and your team.
```
